platform-api/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java

```diff
@@ -213,27 +240,42 @@ public UsersWorkspaceDto startTemporary(WorkspaceConfigDto cfg, @QueryParam("acc
     @POST
     @Path("/{id}/runtime")
     @Produces(APPLICATION_JSON)
+    @RolesAllowed("user")
     public UsersWorkspaceDto startById(@PathParam("id") String workspaceId,
                                        @QueryParam("environment") String envName,
                                        @QueryParam("accountId") String accountId)
             throws ServerException, BadRequestException, NotFoundException, ForbiddenException {
+        ensureUserIsWorkspaceOwner(workspaceId);
+
+        final Map<String, String> params = ImmutableMap.of("accountId", accountId, "workspaceId", workspaceId);
+        permissionManager.checkPermission(START_WORKSPACE, getCurrentUserId(), params);
+
         return injectLinks(DtoConverter.asDto(workspaceManager.startWorkspaceById(workspaceId, envName, accountId)));
     }
 
     @POST
     @Path("/name/{name}/runtime")
     @Produces(APPLICATION_JSON)
+    @RolesAllowed("user")
     public UsersWorkspaceDto startByName(@QueryParam("name") String name,
                                          @QueryParam("environment") String envName,
                                          @QueryParam("accountId") String accountId)
             throws ServerException, BadRequestException, NotFoundException, ForbiddenException {
+        final UsersWorkspace workspace = workspaceManager.getWorkspace(name, getCurrentUserId());
```

Comment:
It would be better to move this line to *ensureUserIsWorkspaceOwner* method because it is part of it.
You can name it as *ensureUserIsWorkspaceOwnerByName*